File: Task/Remove-lines-from-a-file/C-sharp/remove-lines-from-a-file.cs
using System;
using System.IO;
using System.Linq;

public class Rosetta
{
    public static void Main() => RemoveLines("foobar.txt", start: 1, count: 2);

    static void RemoveLines(string filename, int start, int count = 1) =>
        File.WriteAllLines(filename, File.ReadAllLines(filename)
            .Where((line, index) => index < start - 1 || index >= start + count - 1));
}

File: Task/Remove-lines-from-a-file/PascalABC.NET/remove-lines-from-a-file.pas
procedure RemoveLines(fname: string; start: integer; count: integer := 1);
begin
  WriteAllLines(fname,
    ReadAllLines(fname).Where((s,ind) -> ind not in start - 1 .. start + count - 2).ToArray)
end;

begin
  RemoveLines('a2.pas',3,2);
end.
